Fix the flashing of dashboard before onboarding modal

ashwinkhode opened the pull request:
Fixes #6403 

**Replay before fix** - https://app.replay.io/recording/initial-load--638d7723-e238-4574-a519-3944178ac9e4?point=6814889628629178182488377983500923&time=2694.537017205683&hasFrames=true
**Loom Video before fix** - https://www.loom.com/share/00f32b5463774270bf8e9f790091a4ae

**Loom Video after fix** - https://www.loom.com/share/622739417d9045fbac5aba23bdee810e

I tested the fix by changing the nags manually. Seems to be working fine. Please let me know if I missed something related to this issue, I would love to do that :)

jasonLaster commented:
@ashwinkhode might be fun to include a replay of the bug and fix. This way we can see how adjusting the components fixes the issue

jasonLaster commented on packages/accordion/website/pages/_app.js:
```diff
@@ -1,7 +1,7 @@
 }
 
-export default MyApp
+export default MyApp;
```
might be nice to move these prettier changes to their own PR

ashwinkhode replied:
I think if i dont include this [lint fix commit f2f941f](https://github.com/replayio/devtools/pull/7100/commits/f2f941f66f0991929e752b6198d5595217fb0eac), Lint / Trunk check GH action will fail for this PR :(

souporserious replied:
Appreciate your help here! Sorry about our linting, it is a little wonky right now. If you can remove the lint fix commit and keep these changes isolated to the one file I'll get this merged. We can take a look at linting in a follow-up PR and should probably find a time to run it across the whole codebase anyways.

ashwinkhode replied:
@souporserious removed the lint fix commit. Please have a look 🙈

jasonLaster commented on src/ui/components/Library/Library.tsx:
```diff
@@ -41,46 +41,58 @@ function isUnknownWorkspaceId(
+      trackEvent("onboarding.demo_replay_prompt");
+      setModal("first-replay");
+    }
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```
looks cleaner, but how does this fix it?

ashwinkhode replied:
Our onboarding modal logic is executed when dashboard is mounted. I just lifted that logic by one level because we have all the data required available to us in `<LibraryLoader />` component. So as soon as we get the user info, we run the onboarding modal logic in the `<LibraryLoader />` component, previously it was getting executed in `<Library />` component. Does this answer your question? Please correct me if I am wrong anywhere 🤔

ashwinkhode commented:
> @ashwinkhode might be fun to include a replay of the bug and fix. This way we can see how adjusting the components fixes the issue

Hey Jason, I added the replay that was posted in the original issue and also I added my own Loom video recordings for before and after the fix. You can find it in the PR description :)

souporserious reviewed (approved):
This loads much better now ✨. We appreciate your help fixing this, thank you!
